Security/SessionDownloadStrategy.php

```diff
@@ -39,14 +48,34 @@ class SessionDownloadStrategy implements DownloadStrategyInterface
     protected $sessionKey = 'sonata/media/session/times';
 
     /**
-     * @param TranslatorInterface $translator
-     * @param ContainerInterface  $container
-     * @param int                 $times
+     * @var SessionInterface
+     */
+    private $session;
+
+    /**
+     * @param TranslatorInterface                 $translator
+     * @param ContainerInterface|SessionInterface $session
+     * @param int                                 $times
      */
-    public function __construct(TranslatorInterface $translator, ContainerInterface $container, $times)
+    public function __construct(TranslatorInterface $translator, $session, $times)
     {
+        // NEXT_MAJOR : remove this block and set session from parameter.
+        if ($session instanceof ContainerInterface) {
+            @trigger_error(
+                'Using an instance of Symfony\Component\DependencyInjection\ContainerInterface is deprecated since 
+                version 3.x and will be removed in 4.0. 
+                Use Symfony\Component\HttpFoundation\Session\SessionInterface instead.',
+                E_USER_DEPRECATED
+            );
+
+            $this->session = $session->get('session');
+        } elseif ($session instanceof SessionInterface) {
+            $this->session = $session;
+        } else {
+            throw new \InvalidArgumentException();
```

Comment:
Please provide a message : `$session should be an instance of either ContainerInterface or Session Interface`